### ISSUE TITLE
Introduce a linter to encourage using value matching instead of chained equality checks?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2379]
## Preamble

The following two code snippets are equivalent in many scenarios. But the value matching operator (`%in%`) approach is generally preferable for its readability (more concise) and maintainability (as values to check grows, the vector can easily be expanded) over the chained equality checks approach. Of course, this is relevant only if there are more than one comparison in the chain.

## Example that should produce lint

Using chained equality checks

``` r
library(lintr)

'if (x == "a" || x == "b" || x == "c") {
  message("abc")
} else {
  message("not abc")
}' -> code

lint(text = code, linters = all_linters())
```

<sup>Created on 2023-12-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

## Example that shouldn't produce a lint

Using value matching operator

```r
if (x %in% c("a", "b", "c")) {
  message("abc")
} else {
  message("not abc")
}
```

## Edge cases

- If `x` is missing (`NA`), `%in%` will return `FALSE`, whereas the chained equality approach will fail.
- `x` can be a vector in the chain (e.g. `if (any(x == c("a", "b")) || any(x == c("c", "d"))) { ... }`). Such cases won't be relevant for this linter.

**Comment by Bisaloo** [comment]
Is this https://github.com/r-lib/lintr/issues/1875?

**Comment by IndrajeetPatil** [comment]
Indeed! I had a hunch that we had discussed this here before but just couldn't find the issue. 

Closing in favour of that issue.
